Application/Education/Certificate/Generator/Service/DataCertificate/IDataESBD.php

```diff
@@ -0,0 +1,637 @@
+     * @param TblConsumer $tblConsumerCertificate
+     */
+    private static function setEsbdGymHj(Data $Data, TblConsumer $tblConsumerCertificate)
+    {
+
+        $tblCertificate = $Data->createCertificate('Gymnasium Halbjahreszeugnis', '', 'ESBD\EsbdGymHj', $tblConsumerCertificate);
+        if ($tblCertificate) {
+            if ($Data->getTblSchoolTypeGym()) {
+                $Data->updateCertificate($tblCertificate, $Data->getTblCertificateTypeHalfYear(), $Data->getTblSchoolTypeGym());
+                // Update muss erneut ausführbar sein
+//                if (!$Data->getCertificateLevelAllByCertificate($tblCertificate)) {
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypeGym(), '10'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+//                }
+            }
+            // Begrenzung des Bemerkungsfeld
+            $FieldName = 'Remark';
+            if (!$Data->getCertificateFieldByCertificateAndField($tblCertificate, $FieldName)){
+                $Data->createCertificateField($tblCertificate, $FieldName, 600);
+            }
+
+            // Seite 2
+            // Begrenzung der Dialoge
+            self::setESBDSecondPageLength($Data, $tblCertificate);
+        }
+        if ($tblCertificate && !$Data->getCertificateGradeAll($tblCertificate)) {
+            $Data->setCertificateGrade($tblCertificate, 'PN', 1, 1);
+//                $Data->setCertificateGradeAllStandard($tblCertificate);
+        }
+        if ($tblCertificate && !$Data->getCertificateSubjectAll($tblCertificate)) {
+            $Data->setCertificateSubject($tblCertificate, 'DE', 1, 1);
+            $Data->setCertificateSubject($tblCertificate, 'EN', 1, 2);
+            // 1,3 freilassen für Fremdsprache
+            $Data->setCertificateSubject($tblCertificate, 'KU', 1, 4);
+            $Data->setCertificateSubject($tblCertificate, 'MU', 1, 5);
+            $Data->setCertificateSubject($tblCertificate, 'GE', 1, 6);
+            $Data->setCertificateSubject($tblCertificate, 'GRW', 1, 7);
+            $Data->setCertificateSubject($tblCertificate, 'GEO', 1, 8);
+
+            $Data->setCertificateSubject($tblCertificate, 'MA', 2, 1);
+            $Data->setCertificateSubject($tblCertificate, 'BIO', 2, 2);
+            $Data->setCertificateSubject($tblCertificate, 'CH', 2, 3);
+            $Data->setCertificateSubject($tblCertificate, 'PH', 2, 4);
+            $Data->setCertificateSubject($tblCertificate, 'SPO', 2, 5);
+            $Data->setCertificateSubject($tblCertificate, 'RE/e', 2, 6);
+            $Data->setCertificateSubject($tblCertificate, 'TC', 2, 7);
+            $Data->setCertificateSubject($tblCertificate, 'INF', 2, 8);
+        }
+    }
+    /**
+     * @param Data        $Data
+     * @param TblConsumer $tblConsumerCertificate
+     */
+    private static function setEsbdGymJ(Data $Data, TblConsumer $tblConsumerCertificate)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method setEsbdGymJ() has a Cyclomatic Complexity of 15. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5646354)